crates/optimism/txpool/src/error.rs

```diff
@@ -0,0 +1,49 @@
+use reth_optimism_primitives::supervisor::{
+    InteropTxValidatorError, InvalidInboxEntry, SafetyLevel,
+};
+use reth_transaction_pool::error::PoolTransactionError;
+use std::any::Any;
+
+/// Wrapper for [`InteropTxValidatorError`] to implement [`PoolTransactionError`] for it.
+#[derive(thiserror::Error, Debug)]
+#[error(transparent)]
+pub struct InvalidCrossTx(InteropTxValidatorError);
```

Comment:
make this an enum because we need a variant `CrossChainTxPreInterop`. in validate one, if error matches this variant return `InvalidTransactionError::TxTypeNotSupported`. see how it's done for l1.